### PR TITLE
Add rpc to notify channel operation

### PR DESCRIPTION
**Pull request by XuanYang-cn**
- Add NotifyChannelOperation
- Add CheckChannelOperationProgress

See also: #25309

### CODE REVIEW

File: internal/proto/data_coord.proto [hunk 1]
```diff
@@ -694,3 +697,14 @@ message GetFlushStateRequest {
   string collection_name = 4;
   int64 collectionID = 5;
 }
+
+message ChannelOperations {
```
Comment (congqixia) [inline internal/proto/data_coord.proto:701]: Why not follow convention to call it request?

File: internal/datanode/services.go [hunk 2]
```diff
@@ -406,6 +406,18 @@ func (node *DataNode) SyncSegments(ctx context.Context, req *datapb.SyncSegments
 	return merr.Status(nil), nil
 }
 
+func (node *DataNode) NotifyChannelOperation(ctx context.Context, req *datapb.ChannelOperations) (*commonpb.Status, error) {
+	log.Warn("DataNode NotifyChannelOperation is unimplemented")
+	return merr.Status(nil), nil
```
Comment (congqixia) [inline internal/datanode/services.go:411]: NIT: return non-success status here

File: internal/datanode/services.go [hunk 3]
```diff
@@ -406,6 +406,18 @@ func (node *DataNode) SyncSegments(ctx context.Context, req *datapb.SyncSegments
...
+}
+
+func (node *DataNode) CheckChannelOperationProgress(ctx context.Context, req *datapb.ChannelWatchInfo) (*datapb.ChannelOperationProgressResponse, error) {
+	log.Warn("DataNode CheckChannelOperationProgress is unimplemented")
+	return &datapb.ChannelOperationProgressResponse{
+		Status: merr.Status(nil),
```
Comment (congqixia) [inline internal/datanode/services.go:417]: Ditto

File: internal/proto/data_coord.proto (hunk 1 above)
Comment (XuanYang-cn) [reply internal/proto/data_coord.proto:701]: yes,ok

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**